packages/dart_frog_cli/e2e/test/daemon/daemon_domain_test.dart

```diff
@@ -0,0 +1,144 @@
+import 'dart:io';
+
+import 'package:dart_frog_cli/src/daemon/daemon.dart';
+import 'package:path/path.dart' as path;
+import 'package:test/test.dart';
+
+import '../helpers/helpers.dart';
+
+/// Objectives:
+///
+/// * Generate a new Dart Frog project via `dart_frog create`
+/// * Start the daemon
+/// * verify daemon is ready
+/// * send invalid messages
+/// * request daemon.* methods
+void main() {
+  const projectName = 'example';
+  final tempDirectory = Directory.systemTemp.createTempSync();
+  final projectDirectory = Directory(
+    path.join(tempDirectory.path, projectName),
+  );
+
+  late Process daemonProcess;
+
+  late final DaemonStdioHelper daemonStdio;
+
+  setUpAll(() async {
+    await dartFrogCreate(
+      projectName: projectName,
+      directory: tempDirectory,
+    );
+    daemonProcess = await dartFrogDaemonStart(
+      directory: projectDirectory,
+    );
+
+    daemonStdio = DaemonStdioHelper(daemonProcess);
+    addTearDown(() => daemonStdio.dispose());
+  });
+
+  group('daemon domain', () {
+    test('daemon is ready', () async {
+      final readyEvent = await daemonStdio.awaitForDaemonEvent('daemon.ready');
+      expect(readyEvent.params?.keys, containsAll(['version', 'processId']));
+    });
+
+    group('daemon responds to invalid messages', () {
+      test('daemon responds to an invalid message', () async {
+        await daemonStdio.sendStringMessage('ooga boga');
+        final protocolError = await daemonStdio.awaitForDaemonEvent(
+          'daemon.protocolError',
+        );
+        expect(protocolError.params?['message'], 'Not a valid JSON');
```

Comment:
```suggestion
        expect(protocolError.params?['message'], equals('Not a valid JSON'));
```

And the same suggestion to anywhere in the PR where a similar change would apply